src/prommis/uky/costing/models/hydrogen_decrepitation.py

```diff
@@ -174,29 +174,24 @@ def cost_hydrogen_decrepitation_furnace(
             units=units.m**3,
             doc="Internal volume of the furnace in cubic meter",
         )
-        blk.sample_volume = Param(
-            initialize=sample_volume, units=units.m**3, within=Any, mutable=True
-        )
-        blk.sample_density = Param(
-            initialize=sample_density, units=units.kg / (units.m**3), mutable=True
-        )
-        blk.chamber_to_sample_ratio = Param(
-            initialize=chamber_to_sample_ratio, units=units.dimensionless, mutable=True
-        )
+        blk.sample_volume = Param(units=units.m**3, within=Any, mutable=True)
+        blk.sample_volume.set_value(sample_volume)
```

Comment:
What happens if `sample_mass` is not set and `sample_volume` is also not set? I don't think you test this case, and it looks like that would fail.

Reply:
I suggest adding a check before `sample_mass` and `sample_volume` are used that does

`if sample_volume is None and sample_mass is None:`
    `[a graceful exception that fails with an error message]`

Then, add a test that fails on purpose to test the exception. Check the REE costing test cost for how to do this, there are lots of tests that have `with pytest.raises`.

Reply:
@Akintomiwaojo As @bpaul4 indicated, if you have cases like this where it is required for the user to provide certain information, then you need to have some sort of check in the code to catch cases where that is not the case and return a useful error message to the user. Otherwise, someone will try to provide different information (or insufficient information) and hte model will fail to solve but the user will have no idea of why.